ipv8/messaging/anonymization/community.py

```diff
@@ -928,3 +933,42 @@ async def dht_peer_lookup(self, mid, peer=None):
             await self.dht_provider.peer_lookup(mid, peer)
         else:
             self.logger.error("Need a DHT provider to connect to a peer using the DHT")
+
+    def send_test_request(self, circuit, request_size=0, response_size=0):
+        cache = self.request_cache.add(TestRequestCache(self, circuit))
```

Comment:
I don't think anyone will send a `send_test_request` while shutting down, but just for completeness: `request_cache.add()` can return `None`.

An easy way to avoid this would be:

```python
cache = TestRequestCache(self, circuit)
self.request_cache.add(cache)
```

Reply:
Normally, I would fix this by calling `RequestCache.unload` until after the community is unloaded. However, in this case, the `RequestCache.add` is not called from within a message handler. So, you're right. I'll change it.